node/router/src/handshake.rs

```diff
@@ -264,6 +264,12 @@ impl<N: Network> Router<N> {
         if self.is_connected(&peer_ip) {
             bail!("Dropping connection request from '{peer_ip}' (already connected)")
         }
+        // Only allow trusted peers to connect if we are a validator
+        // (unless allow_external_peers is set)
+        let is_validator = self.node_type().is_validator();
```

Comment:
nit: is_validator is no longer necessary as part of the check (because allow_external_peers takes precedence)